### PR TITLE
fix(iot-dev): Fix amqp layer not checking for local errors during err…

### PR DESCRIPTION
**Pull request by timtay-microsoft**
…or mapping

The SDK was only checking for remote errors previously

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java [hunk 1]
```diff
@@ -1229,20 +1229,34 @@ private TransportException getTransportExceptionFromEvent(Event event)
...
-        String connectionErrorDescription = event.getConnection() != null && event.getConnection().getRemoteCondition() != null && event.getConnection().getRemoteCondition().getDescription() != null ? event.getConnection().getRemoteCondition().getDescription() : "";
-        String linkErrorDescription = event.getLink() != null && event.getLink().getRemoteCondition() != null && event.getLink().getRemoteCondition().getDescription() != null ? event.getLink().getRemoteCondition().getDescription() : "";
-        String transportErrorDescription = event.getTransport() != null && event.getTransport().getRemoteCondition() != null && event.getTransport().getRemoteCondition().getDescription() != null ? event.getTransport().getRemoteCondition().getDescription() : "";
-
+        String senderErrorRemote = event.getSender() != null && event.getSender().getRemoteCondition() != null && event.getSender().getRemoteCondition().getCondition() != null ? event.getSender().getRemoteCondition().getCondition().toString() : "";
```
Comment (davilu) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java]: Please create a function `<T> String nullToBlank(T value)`

Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java]: I opted to use the Endpoint class that proton defines to generalize searching through these sender/receiver/link/transport classes and their errors

**Pull request merged**